swift-sdk/Internal/ApiClient.swift

```diff
@@ -48,12 +48,13 @@ protocol AuthProvider: class {
     var auth: Auth { get }
 }
 
-class ApiClient : ApiClientProtocol {
-    init(apiKey: String, authProvider: AuthProvider, endPoint: String, networkSession: NetworkSessionProtocol) {
+class ApiClient: ApiClientProtocol {
+    init(apiKey: String, authProvider: AuthProvider, endPoint: String, networkSession: NetworkSessionProtocol, deviceId: String) {
         self.apiKey = apiKey
         self.authProvider = authProvider
         self.endPoint = endPoint
         self.networkSession = networkSession
+        self.deviceId = deviceId
```

Comment:
`deviceId` seems too specific to add here.. i.e. even the auth part is abstracted into a provider. What do you think about either passing deviceId explicitly (like it is now in `createRegisterTokenRequest`), or creating another provider?
`deviceId` is also used in `createRegisterTokenRequest` so whatever we do with `deviceId`, it should be the same in `createRegisterTokenRequest` and `track*`.